### ISSUE TITLE
Issue with "," in the header

### ISSUE THREAD

**Issue by santiagoluisfuse** [mholt/PapaParse#600]
For some reason when I try to parse this CSV file the detected delimiter is ",", should be ","

SKU (alta dejar vacío, en modificación obligatorio);Nombre (100 caracteres o menos, obligatorio);Descripción del producto (obligatorio);Marca (alta previa, obligatorio);UPC (número entero, 20 digitos máximo, obligatorio);Precio regular (obligatorio);Precio con descuento;Precio válido hasta (año-mes-día);URL's de imágenes (separados por comas, obligatorio);Alto (cm, obligatorio);Largo (cm, obligatorio);Peso (gr, obligatorio);Ancho (cm, obligatorio)
1542822078456;Santa test 3;Corbata;El Burgues;709;90.0;80.0;2018-11-11;https://fravega.vteximg.com.br/arquivos/ids/2661035-1000-1000/501722_1.jpg;10.0;10.0;1000.0;10.0
1542822078493;Santa test 4;Corbata;El Burgues;800;90.0;80.0;2018-11-11;https://fravega.vteximg.com.br/arquivos/ids/2661035-1000-1000/501722_1.jpg;10.0;10.0;1000.0;10.0

**Comment by pokoli** [comment]
hi @santiagoluisfuse, 

I see there is a loot of , used on the first line, so thay may be the reason why PapaParse detects it as delimiter. 

Why you don't force the deleititer to be ';' instead of autodetecting it?

**Comment by dboskovic** [comment]
Yeah, those commas are definitely confusing it. I do think this is a great edge-case to ask for contribution for. I think our auto-detection is still pretty primitive and could benefit from some more iteration. I think it's possible we'd see the inverse of this problem where a header with a lot of pipes or semicolons in the values would get read the wrong way.

## Help wanted summary
- [ ] Add a test using an anonymized version of the above data
- [ ] Update our auto-detection logic to detect the *right* delimiter even with commas in the header values
- [ ] Make sure existing tests don't break

## How to contribute
https://github.com/mholt/PapaParse#contributing

**Comment by heatherkellyucl** [comment]
Just had the reverse case where something previously parsed as comma-separated was erroneously picked up as pipe-separated. (The first line only had commas but the rest contain a lot of pipes).

We have started setting the delimiter manually everywhere, but it took a while to work out why everything was failing!

Anonymised example:
```
,C Time,W Time
xxx1356 | A Name | kli,2964067,13797
xxx1633 | B Surname | EFG | EFG | EFG_heer_kip | Jeiner_FF,2114779,29200
fffaryo | C Nominator | SWA | SWA_eeer_Feeer | Reiner_LLL_erw,1492251,9812
cccadns | D Surname | hyt,539147,9520
xxx1701 | Edwin Name | GYH | GYH_ghuty_der | Jeiner_SWS_bjm,26286,853
xxx1935 | Fred Surname | Moorner_Let | Jeiner_fruit | Jeiner_JKL_fru,9546,98
```
It was being run as `Papa.parse(csv, {skipEmptyLines": true}).data;` with no header.
We have been using the PapaParse master.
